engine/access/rpc/backend/backend_transactions_test.go

```diff
@@ -0,0 +1,329 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/onflow/flow/protobuf/go/flow/access"
+	"github.com/onflow/flow/protobuf/go/flow/entities"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/state/protocol"
+	bprotocol "github.com/onflow/flow-go/state/protocol/badger"
+	"github.com/onflow/flow-go/state/protocol/util"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func (suite *Suite) WithPreConfiguredState(f func(snap protocol.Snapshot)) {
+	identities := unittest.CompleteIdentitySet()
+	rootSnapshot := unittest.RootSnapshotFixture(identities)
+	util.RunWithFullProtocolState(suite.T(), rootSnapshot, func(db *badger.DB, state *bprotocol.ParticipantState) {
+		epochBuilder := unittest.NewEpochBuilder(suite.T(), state)
+
+		epochBuilder.
+			BuildEpoch().
+			CompleteEpoch()
+
+		// get heights of each phase in built epochs
+		epoch1, ok := epochBuilder.EpochHeights(1)
+		require.True(suite.T(), ok)
+
+		// setup AtHeight mock returns for state
+		for _, height := range epoch1.Range() {
+			suite.state.On("AtHeisght", epoch1.Range()).Return(state.AtHeight(height))
+		}
+
+		snap := state.AtHeight(epoch1.Range()[0])
+		suite.state.On("Final").Return(snap).Once()
+		suite.communicator.On("CallAvailableNode",
+			mock.Anything,
+			mock.Anything,
+			mock.Anything).
+			Return(nil)
+
+		f(snap)
+	})
+
+}
+
+func (suite *Suite) TestGetTransactionResultReturnsUnknown() {
+	suite.WithPreConfiguredState(func(snap protocol.Snapshot) {
+		block := unittest.BlockFixture()
+		tbody := unittest.TransactionBodyFixture()
+		tx := unittest.TransactionFixture()
+		tx.TransactionBody = tbody
+
+		coll := flow.CollectionFromTransactions([]*flow.Transaction{&tx})
+		suite.state.On("AtBlockID", block.ID()).Return(snap, nil).Once()
+
+		suite.transactions.
+			On("ByID", tx.ID()).
+			Return(nil, storage.ErrNotFound)
+
+		suite.blocks.
+			On("ByID", block.ID()).
+			Return(&block, nil).
+			Once()
+
+		receipt := unittest.ExecutionReceiptFixture()
+		identity := unittest.IdentityFixture()
```

Comment:
doesn't look like this is used in any of the tests. Does it need to be added somewhere or can it be removed?

Reply:
yes, cleaned up )

Reply:
https://github.com/onflow/flow-go/pull/4598/commits/0d51ec31da7b738a72309c263777f27b87200ff2